contrib/excluder/excluder-template

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#
+# This script will either enable, or disable, the updating 
+#  of openshift rpms via yum
+#
+##############
+# VARIABLES
+##############
+CONF_FILE="@@CONF_FILE-VARIABLE@@"
+PACKAGE_LIST="@@PACKAGE_LIST-VARIABLE@@"
+
+##############
+# HELP
+##############
+usage() {
+  echo "Usage: ${0} <exclude | unexclude>" >&2
+  echo >&2
+  echo "exclude | disable" >&2                                                                                         
+  echo "  exclude packages via an excludes= line in ${CONF_FILE}" >&2                                    
+  echo "  This disables updates for ${PACKAGE_LIST} packages" >&2
+  echo >&2
+  echo "unexclude | enable" >&2
+  echo "  removes ${PACKAGE_LIST} from the excludes= line in ${CONF_FILE}" >&2
+  echo "  This re-enables updates for packages" >&2
+  echo >&2
+  popd &>/dev/null
+  exit 1
+}
+
+##############
+# FUNCTIONS
+##############
+exclude_packages() {
+  # Ensure there is an exclude line
+  if ! grep -q exclude= ${CONF_FILE} ; then
+    echo "Adding exclude= to ${CONF_FILE}"
+    sed 's|\(obsoletes=.*\)|\1\nexcludes=|' ${CONF_FILE}
+  fi
+  # Add each package to the exclude line
+  for package in ${PACKAGE_LIST}
+  do
+    if ! grep exclude= ${CONF_FILE} | grep -F -q " ${package} " ; then
+      sed -i "s|exclude=|exclude= ${package} |" ${CONF_FILE}
+    fi
+  done
+}
+
+unexclude_packages() {
+  # Remove each package to the exclude line
+  for package in ${PACKAGE_LIST}
+  do
+    if grep exclude= ${CONF_FILE} | grep -F -q " ${package} " ; then
+      slashpackage=$(echo ${package} | sed 's|\*|\\\*|g')
+      sed -i "s| ${slashpackage} ||" ${CONF_FILE}
+    fi
+  done
+}
+
+##############
+# MAIN PROGRAM
+##############
+case "$1" in
+  exclude | disable )
+    exclude_packages
+    ;;
+  unexclude | enable )
+    unexclude_packages
+    ;;
+  * )
+    echo "Unknown Option: ${1}"
+    echo
+    usage
+    ;;
+esac
+
+exit 0
```

Comment:
Correct.
Since this is a script that get's run by rpm, I want to make sure it exits with 0 if it get's to that point.  If any other exit code pops up, it breaks the rpm install.